fuse/nodefs/fsops.go

```diff
@@ -70,6 +70,8 @@ func (c *FileSystemConnector) lookupMountUpdate(out *fuse.Attr, mount *fileSyste
 
 // internalLookup - execute a lookup without affecting NodeId reference counts
 func (c *FileSystemConnector) internalLookup(out *fuse.Attr, parent *Inode, name string, header *fuse.InHeader) (node *Inode, code fuse.Status) {
+	c.forgetMu.RLock()
```

Comment:
No, the original code did not remove the forgotten node at all from the tree. I could not because it does not have parent pointers. It removed all the children, but not the node itself. By the way it also never called OnForget on the node itself.